gulpfile.js

```diff
@@ -38,7 +38,7 @@ gulp.task('ghpages', 'Deploy documentation to gh-pages', ['nojekyll'], function
 
 gulp.task("docs", 'Compile documentation from src code', function () {
   return gulp
-    .src(["src/**/*.ts"])
+    .src(["typings/globals/es6-promise/index.d.ts", "src/**/*.ts"])
```

Comment:
It sounds like there are two different issues here.

1. Typedoc doesn't support union types
2. External .d.ts files need to be included
(es6-promise, wpmp, hpm, router, models)

The second issue is indepedent